src/FontDescriptor.h

```diff
@@ -46,161 +46,23 @@ struct FontDescriptor {
-
-  ~FontDescriptor() {
-    if (path)
-      delete path;
-
-    if (postscriptName)
-      delete postscriptName;
-
-    if (family)
-      delete family;
-
-    if (style)
-      delete style;
-
-    postscriptName = NULL;
-    family = NULL;
-    style = NULL;
-  }
-
-  Local<Object> toJSObject() {
-    Nan::EscapableHandleScope scope;
-    Local<Object> res = Nan::New<Object>();
-    if (path) {
-      Nan::Set(res, Nan::New("path").ToLocalChecked(), Nan::New<String>(path).ToLocalChecked());
-    }
-
-    if (postscriptName) {
-      Nan::Set(res, Nan::New("postscriptName").ToLocalChecked(), Nan::New<String>(postscriptName).ToLocalChecked());
-    }
-
-    if (family) {
-      Nan::Set(res, Nan::New("family").ToLocalChecked(), Nan::New<String>(family).ToLocalChecked());
-    }
-
-    if (style) {
-      Nan::Set(res, Nan::New("style").ToLocalChecked(), Nan::New<String>(style).ToLocalChecked());
-    }
-
-    Nan::Set(res, Nan::New("weight").ToLocalChecked(), Nan::New<Number>(weight));
-    Nan::Set(res, Nan::New("width").ToLocalChecked(), Nan::New<Number>(width));
-    Nan::Set(res, Nan::New("italic").ToLocalChecked(), Nan::New<v8::Boolean>(italic));
-    Nan::Set(res, Nan::New("monospace").ToLocalChecked(), Nan::New<v8::Boolean>(monospace));
-    return scope.Escape(res);
-  }
-
-private:
-  char *copyString(const char *input) {
-    if (!input)
-      return NULL;
-
-    char *str = new char[strlen(input) + 1];
-    strcpy(str, input);
-    return str;
-  }
-
-  char *getString(Local<Object> obj, const char *name) {
-    Nan::HandleScope scope;
-    Nan::MaybeLocal<v8::Value> maybeValue = Nan::Get(obj, Nan::New(name).ToLocalChecked());
-
-    if (maybeValue.IsEmpty()) {
-      return NULL;
-    }
-
-    v8::Local<v8::Value> value = maybeValue.ToLocalChecked();
-    if (value->IsString()) {
-      return copyString(*Nan::Utf8String(Nan::To<v8::String>(value).ToLocalChecked()));
-    }
-
-    return NULL;
-  }
-
-  int getNumber(Local<Object> obj, const char *name) {
-    Nan::HandleScope scope;
-    Nan::MaybeLocal<v8::Value> maybeValue = Nan::Get(obj, Nan::New(name).ToLocalChecked());
-
-    if (maybeValue.IsEmpty()) {
-      return 0;
-    }
-
-    v8::Local<v8::Value> value = maybeValue.ToLocalChecked();
-    if (value->IsNumber()) {
-      return Nan::To<int>(value).FromJust();
-    }
-
-    return 0;
-  }
-
-  bool getBool(Local<Object> obj, const char *name) {
-    Nan::HandleScope scope;
-    Nan::MaybeLocal<v8::Value> maybeValue = Nan::Get(obj, Nan::New(name).ToLocalChecked());
-
-    if (maybeValue.IsEmpty()) {
-      return false;
-    }
-
-    v8::Local<v8::Value> value = maybeValue.ToLocalChecked();
-    if (value->IsBoolean()) {
-      return Nan::To<bool>(value).FromJust();
-    }
-
-    return false;
-  }
-};
-
-class ResultSet : public std::vector<FontDescriptor *> {
-public:
-  ~ResultSet() {
-    for (ResultSet::iterator it = this->begin(); it != this->end(); it++) {
-      delete *it;
-    }
-  }
+  FontDescriptor();
```

Comment:
default copy constructor and assignment will copy pointers. Probably we don't want that. By adding them as deleted we make sure that that's not happening. Also delete move constructor and assignment to make sure the code is not calling them unintentionally.
```suggestion
  FontDescriptor (const FontDescriptor&) = delete;
  FontDescriptor& operator= (const FontDescriptor&) = delete;
  FontDescriptor (FontDescriptor&&) = delete;
  FontDescriptor& operator= (FontDescriptor&&) = delete;
```